android/src/main/java/io/ably/lib/push/ActivationStateMachine.java

```diff
@@ -142,6 +140,9 @@ public void execute(HttpScheduler http, Callback<JsonObject> callback) throws Ab
                             if(ably.options.pushFullWait) {
                                 params = Param.push(null, "fullWait", "true");
                             }
+                            if(ably.options.addRequestIds) { // RSC7c
```

Comment:
While I recognise that this repetition was already there (re `pushFullWait` > `fullWait` lines above), could we please DRY this up by perhaps taking this params building code and moving it somewhere common. I think it would be appropriate to do that in the scope of this pull request as the size of this common code now doubles.

Reply:
I've pulled common logic out, but not sure if to the best place. Please let me know if you have a better idea.

Reply:
I see you (https://github.com/ably/ably-java/pull/678/commits/89ecf93d93076f20c54eef931f324b20bd445068).

I think the location of this new code is entirely appropriate. What I would like to see now, however, for this new method is:

1. API commentary
2. Unit test(s)